src-java/testing/test-library/src/main/java/org/openkilda/testing/service/northbound/IslandNorthboundServiceV2.java

```diff
@@ -47,7 +50,21 @@ public class IslandNorthboundServiceV2 extends NorthboundServiceV2Impl {
     public List<FlowResponseV2> getAllFlows() {
         List<FlowResponseV2> result = super.getAllFlows();
         return result.stream().filter(flow -> switchIds.stream().anyMatch(swId ->
-                flow.getSource().getSwitchId().equals(swId) || flow.getDestination().getSwitchId().equals(swId)))
+                flow.getSource().getSwitchId().equals(swId) && Objects.isNull(flow.getYFlowId())))
```

Comment:
There is no need to check both source and destination as flow is deployed inside one topology lab.
As yFlow's sub-flows are regular flows, we need to exclude them from the list of regular flows.